tetrateci/tsb-util/k8s-objects/traffic-gen.yaml

```diff
@@ -0,0 +1,58 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: trafficgenerator
+  name: trafficgenerator
+  namespace: {{ ns }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: trafficgenerator
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false"
+      labels:
+        app: trafficgenerator
+    spec:
+      containers:
+      - args:
+        - /bin/sh
+        - -c
+        - |
+          while [[ ${IP} == "" ]]; do
+            IP=$(kubectl get service -n {{ ns }} istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+            sleep 10
+          done
+          while :; do
+            curl "https://{{ hostname }}/productpage" --resolve "{{ hostname }}:$IP" --cacert /etc/bookinfo/bookinfo-ca.crt #>/dev/null 2>&1
+            sleep 10
+          done
+        image: bitnami/kubectl
+        imagePullPolicy: IfNotPresent
+        name: trafficgenerator
+        resources: {}
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+        volumeMounts:
+        - mountPath: /etc/bookinfo
+          name: bookinfo
+          readOnly: true
+      dnsPolicy: ClusterFirst
+      restartPolicy: Always
+      schedulerName: default-scheduler
+      securityContext: {}
+      terminationGracePeriodSeconds: 30
+      volumes:
+      - name: bookinfo
+        secret:
+          defaultMode: 420
+          secretName: {{ secretName }}
```

Comment:
This is a different secret which is a generic one. Where and how are you creating this secret? Refer to my branch for details

Reply:
isnt it the same secret what we use in the gateway?